concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/30 Save, Cancel, and Custom Buttons.md

```diff
@@ -0,0 +1,23 @@
+Form and popup editing modes include built-in Save and Cancel buttons. If you want to execute custom code on click one of these buttons, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler correspondingly. Assign `true` to the **e.cancel** parameter to cancel the Save or Cancel operations and keep the form or popup opened.
+
+Specify the **editing.texts**.[cancelRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#cancelRowChanges) and **editing.texts**.[saveRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#saveRowChanges) properties to change texts of the Save and Cancel buttons.
+
+To customize the Save and Cancel buttons, re-define them:
+
+- Popup edit mode     
+Add the **editing**.[popup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#popup) object to the DataGrid configuration and declare the Save, Cancel and custom buttons in the **popup**.[toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) array.
```

Comment:
>  and declare **the** Save, Cancel and custom buttons

Again. Plus, no Oxford comma.

```suggestion
Add the **editing**.[popup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#popup) object to the DataGrid configuration and declare Save, Cancel, and custom buttons in the **popup**.[toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) array.
```

P.S. "Configuration" looks like an adjective rather than noun. Did you mean "configuration file" in this case? Second. You use a definite article when you say "add the X object to ..." Since you mean an object (instance), you may want to use an indefinite article case here. If this object is something like "one of the kind" -- then you are probably right. Still, you may want to double-check this fact.